src/rgw/rgw_rados.cc

```diff
@@ -1204,6 +1204,8 @@ int RGWPeriod::reflect()
       ldout(cct, 0) << "ERROR: failed to store zonegroup info for zonegroup=" << iter.first << ": " << cpp_strerror(-r) << dendl;
       return r;
     }
+    // set as default if no default exists
+    zg.set_as_default(true);
```

Comment:
sure. i think i'll also add a ldout message if it successfully changes the default